### PR TITLE
add option to use rails ajax

### PR DESCRIPTION
**Pull request by JoeMilsom**
why: when we are polling we want to have javascript to be executed in the response. rails ajax makes it so we can execute it without a CSP error

### CODE REVIEW

File: app/views/render_async/_request_rails_ajax.js.erb [hunk 1]
```diff
@@ -0,0 +1,141 @@
+if (window.jQuery) {
```
Comment (gingermusketeer) [inline app/views/render_async/_request_rails_ajax.js.erb:1]: Might be good to change this to `window.Rails`.

File: app/views/render_async/_request_rails_ajax.js.erb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      Rails.ajax({
+        url: '<%= path.html_safe %>',
+        type: '<%= method %>',
+        data: "<%= escape_javascript(data.to_s.html_safe) %>",
+        headers: headers,
+        onSuccess: onSuccess,
+        onError: onError
+      })
+
+      function onSuccess(response){
```
Comment (gingermusketeer) [inline app/views/render_async/_request_rails_ajax.js.erb:48]: If the response is application/javascript should this code run?

File: app/views/render_async/_request_rails_ajax.js.erb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      });
+    }
+
+    _runAfterDocumentLoaded(_setUpToggle);
+    <% elsif !toggle %>
+    _runAfterDocumentLoaded(_renderAsyncFunction)
+    <% end %>
+  }(jQuery));
+} else {
+  console.warn("Looks like you've enabled jQuery for render_async, but jQuery is not defined on the window object");
```
Comment (gingermusketeer) [inline app/views/render_async/_request_rails_ajax.js.erb:140]: Be good to update this as well.

File: lib/render_async/configuration.rb [hunk 4]
```diff
@@ -1,10 +1,11 @@
 module RenderAsync
   class Configuration
-    attr_accessor :jquery, :turbolinks
+    attr_accessor :jquery, :turbolinks, :rails_ajax
```
Comment (gingermusketeer) [inline lib/render_async/configuration.rb:3]: perhaps this option should be called `rails_ujs` to match the name of the library we are using.

File: app/views/render_async/_request_rails_ajax.js.erb (hunk 2 above)
Comment (JoeMilsom) [reply app/views/render_async/_request_rails_ajax.js.erb]: it will fail gracefully due to js, but i added a check in anyway

**Comment by JoeMilsom** [comment]
kind of think that these changes we are making to this library kind of fall outside the intended purpose of the library. maybe we could create a js script to do this instead

File: app/views/render_async/_request_rails_ujs.js.erb [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+          _renderAsyncFunction();
+        }
+      });
+    }
+
+    _runAfterDocumentLoaded(_setUpToggle);
+    <% elsif !toggle %>
+    _runAfterDocumentLoaded(_renderAsyncFunction)
+    <% end %>
+  }(jQuery));
```
Comment (gingermusketeer) [inline app/views/render_async/_request_rails_ujs.js.erb:144]: I guess this is not needed?

**Comment by JoeMilsom** [comment]
> Looks good. Does it work locally?

sure does buddy

File: app/views/render_async/_request_rails_ujs.js.erb [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        $(document).ready(callback);
+        <% end %>
+      }
+    }
+
+    function _makeRequest(currentRetryCount) {
+      var headers = <%= headers.to_json.html_safe %>;
+      var csrfTokenElement = document.querySelector('meta[name="csrf-token"]')
+      if (csrfTokenElement)
+        headers['X-CSRF-Token'] = csrfTokenElement.content
```
Comment (gingermusketeer) [inline app/views/render_async/_request_rails_ujs.js.erb:37]: Does `Rails.ajax` do this already?

**Pull request merged**